### PR TITLE
Terraform module to provision a kafka proxy

### PR DESCRIPTION
**Pull request by damondouglas**
This PR automates [this guide](https://cloud.google.com/kubernetes-engine/docs/tutorials/private-cluster-bastion) using a terraform module to provision a compute engine instance bastion host to act as a proxy to a private Kafka cluster. See README committed in this PR for description, prerequisites, requirements, and usage. The goal is so that the same Kafka cluster can be used for local development/testing as well as for remote usage in a pipeline such as Dataflow.

Essentially, the solution creates a bastion host compute engine instance and outputs the gcloud command needed to tunnel traffic to the developer's local machine.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by shunping** [comment]
Sorry @damondouglas. I am not able to provide a comprehensive review on this PR as my understanding on Terraform is limited. Do you have a reviewer in mind that can help with this?

File: .test-infra/kafka/proxy/common.tfvars
Comment (KevinGG) [inline .test-infra/kafka/proxy/common.tfvars:1]: Do tfvars files need to be checked in? Can these be set as the default values of the variables?

**Review by KevinGG — APPROVED** [review]
LGTM except a nit.

File: .test-infra/kafka/proxy/common.tfvars
Comment (damondouglas) [reply .test-infra/kafka/proxy/common.tfvars:1]: In my experience the use of default values masks the requirement of these variables. Version controlling the tfvars allows for opinionated defaults while maintaining an emphasis on their required state. One can use the `-var-file` flag multiple times.

**Pull request merged**